pkg/build/admission/defaults/admission.go

```diff
@@ -130,6 +141,18 @@ func (a *buildDefaults) applyBuildDefaults(build *buildapi.Build) {
 			build.Spec.Source.Git.NoProxy = &t
 		}
 	}
+
+	//apply default source secret if one set after all validation
+	//BUG: build fails because secret is not mounted... where to mount it?
```

Comment:
apparently im doing something wrong:
```
[mangis@mj amd64]$ oc project --config=config/master/admin.kubeconfig
Using project "test" on server "https://192.168.1.116:8443".
```

secret : 
```
[mangis@mj amd64]$ oc get secrets --config=config/master/admin.kubeconfig
NAME                       TYPE                                  DATA      AGE
scmsecret                  Opaque                                1         10m
```

after build is executed:
```
error: cannot setup source secret: no auth handler was found for secrets in /tmp/tmpsecret195217705
```
and pod describe does not show mounted secret:
```
oc describe pod ruby-sample-build-2-build --config=config/master/admin.kubeconfig
Name:			ruby-sample-build-2-build
Namespace:		test
Security Policy:	privileged
Node:			mj/192.168.1.116
Start Time:		Mon, 17 Oct 2016 19:47:50 +0100
Labels:			openshift.io/build.name=ruby-sample-build-2
Status:			Failed
IP:			172.17.0.2
Controllers:		<none>
Containers:
  sti-build:
    Container ID:	docker://3cd0a689909b5f0f9fad3f8c7f24ae4ecc4d755b76315039d689de6adaa63b66
    Image:		openshift/origin-sti-builder:v1.4.0-alpha.0
    Image ID:		docker://sha256:70676b107f3957c3bf8913f55e06a0385026e632cf998c78f5e781159ab870ef
    Port:		
    Args:
      --loglevel=0
    State:		Terminated
      Reason:		Error
      Exit Code:	1
      Started:		Mon, 17 Oct 2016 19:47:54 +0100
      Finished:		Mon, 17 Oct 2016 19:47:54 +0100
    Ready:		False
    Restart Count:	0
    Volume Mounts:
      /var/run/docker.sock from docker-socket (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from builder-token-u01il (ro)
      /var/run/secrets/openshift.io/push from builder-dockercfg-tvzes-push (ro)
    Environment Variables:
      BUILD:	{"kind":"Build","apiVersion":"v1","metadata":{"name":"ruby-sample-build-2","namespace":"test","selfLink":"/oapi/v1/namespaces/test/builds/ruby-sample-build-2","uid":"348d5e97-949a-11e6-b21f-507b9dbf3051","resourceVersion":"2362","creationTimestamp":"2016-10-17T18:47:50Z","labels":{"app":"ruby-helloworld-sample","buildconfig":"ruby-sample-build","name":"ruby-sample-build","openshift.io/build-config.name":"ruby-sample-build","openshift.io/build.start-policy":"Serial","template":"application-template-stibuild"},"annotations":{"openshift.io/build-config.name":"ruby-sample-build","openshift.io/build.number":"2"}},"spec":{"serviceAccount":"builder","source":{"type":"Git","git":{"uri":"https://github.com/openshift/ruby-hello-world.git","httpProxy":"http://my.proxy:8080"},"sourceSecret":{"name":"scmsecret"}},"strategy":{"type":"Source","sourceStrategy":{"from":{"kind":"DockerImage","name":"centos/ruby-22-centos7@sha256:bab948dcca9f9f86014a25c0e0020438047bbed235cfbc5cfc375922ec04ddf3"},"env":[{"name":"EXAMPLE","value":"sample-app"}]}},"output":{"to":{"kind":"DockerImage","name":"172.30.161.64:5000/test/origin-ruby-sample:latest"},"pushSecret":{"name":"builder-dockercfg-tvzes"}},"resources":{},"postCommit":{"args":["bundle","exec","rake","test"]},"triggeredBy":[{"message":"Manually triggered"}]},"status":{"phase":"New","outputDockerImageReference":"172.30.161.64:5000/test/origin-ruby-sample:latest","config":{"kind":"BuildConfig","namespace":"test","name":"ruby-sample-build"}}}

      SOURCE_REPOSITORY:	https://github.com/openshift/ruby-hello-world.git
      SOURCE_URI:		https://github.com/openshift/ruby-hello-world.git
      ORIGIN_VERSION:		v1.4.0-alpha.0+9a0f45b-dirty
      ALLOWED_UIDS:		1-
      DROP_CAPS:		KILL,MKNOD,SETGID,SETUID,SYS_CHROOT
      PUSH_DOCKERCFG_PATH:	/var/run/secrets/openshift.io/push
Conditions:
  Type		Status
  Initialized 	True 
  Ready 	False 
  PodScheduled 	True 
Volumes:
  docker-socket:
    Type:	HostPath (bare host directory volume)
    Path:	/var/run/docker.sock
  builder-dockercfg-tvzes-push:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	builder-dockercfg-tvzes
  builder-token-u01il:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	builder-token-u01il
QoS Tier:	BestEffort
Events:
  FirstSeen	LastSeen	Count	From			SubobjectPath			Type		Reason		Message
  ---------	--------	-----	----			-------------			--------	------		-------
  7m		7m		1	{default-scheduler }					Normal		Scheduled	Successfully assigned ruby-sample-build-2-build to mj
  7m		7m		1	{kubelet mj}		spec.containers{sti-build}	Normal		Pulled		Container image "openshift/origin-sti-builder:v1.4.0-alpha.0" already present on machine
  7m		7m		1	{kubelet mj}		spec.containers{sti-build}	Normal		Created		Created container with docker id 3cd0a689909b; Security:[seccomp=unconfined]
  7m		7m		1	{kubelet mj}		spec.containers{sti-build}	Normal		Started		Started container with docker id 3cd0a689909b
```

but source secret is available in description:
```
sourceSecret":{"name":"scmsecret"}}
```

any hints?